nixos/modules/services/hardware/nvidia-container-toolkit/default.nix

```diff
@@ -129,137 +128,191 @@
           '';
         };
       };
-
     };
 
-  config = lib.mkIf config.hardware.nvidia-container-toolkit.enable {
-    assertions = [
-      {
-        assertion =
-          config.hardware.nvidia.datacenter.enable
-          || lib.elem "nvidia" config.services.xserver.videoDrivers
-          || config.hardware.nvidia-container-toolkit.suppressNvidiaDriverAssertion;
-        message = ''`nvidia-container-toolkit` requires nvidia drivers: set `hardware.nvidia.datacenter.enable`, add "nvidia" to `services.xserver.videoDrivers`, or set `hardware.nvidia-container-toolkit.suppressNvidiaDriverAssertion` if the driver is provided by another NixOS module (e.g. from NixOS-WSL)'';
-      }
-      {
-        assertion =
-          ((builtins.length config.hardware.nvidia-container-toolkit.csv-files) > 0)
-          -> config.hardware.nvidia-container-toolkit.discovery-mode == "csv";
-        message = ''When CSV files are provided, `config.hardware.nvidia-container-toolkit.discovery-mode` has to be set to `csv`.'';
-      }
-    ];
+  config = lib.mkMerge [
+    (lib.mkIf config.virtualisation.docker.enableNvidia {
```

Comment:
Isn't this option deprecated? I thought `hardware.nvidia-container-toolkit.enable` was the new one.

Reply:
It is. But people still follow tutorials that point to the old way. This is just a way to be nicer to them in NixOS. I hope we can eventually remove this and settle on CDI for good at some point.

Reply:
Ah, I think I see -- looking through the modules, I see configurations which are conditioned on that value specifically and that it can have a different value than `hardware.nvidia-container-toolkit.enable` (there's no assertion which requires them to be equal).